### PR TITLE
Fhac 1312 - Refactored code to resolve Sonar findings for CXF_POC

### PR DESCRIPTION
**Pull request by mikhelamdex**
Refactoring the following classes in gov.hhs.fha.nhinc.callback.cxf.largefile package:
CONNECTTimestamp.java
TimestampValidator.java

@mhpnguyen @jsmith2 @sjarral112 - Please review

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java [hunk 1]
```diff
@@ -89,38 +86,61 @@ public boolean isExpired(Date invocationDate) {
...
+        }
+    }
+
+    /**
+     * Checks if created time is in the past
+     */
+    private boolean verifyCreatedBefore(Date date) {
+        // Validate the time it took the message to travel
+        if (getCreated() != null && getCreated().before(date)) {
+            if (LOG.isDebugEnabled()) {
```
Comment (christophermay07) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java]: Stylistic suggestion -- usually, "isDebugEnabled()" should be reserved for instances where there is a costly calculation going into the log statement.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java [hunk 2]
```diff
@@ -89,38 +86,61 @@ public boolean isExpired(Date invocationDate) {
...
         invocationTime -= timeToLive * 1000;
-        invocationDate.setTime(invocationTime);
+        date.setTime(invocationTime);
 
         // Validate the time it took the message to travel
-        if (getCreated() != null && getCreated().before(invocationDate)) {
+        if (!verifyCreatedBefore(date)) {
             if (LOG.isDebugEnabled()) {
-                LOG.debug("Validation of Timestamp: The message was created too long ago");
+                LOG.debug("Validation of Timestamp: Everything is ok");
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java]: this comment and condition doesn't look right.  Please verify the logic

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java (hunk 1 above)
Comment (mhpnguyen) [reply Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java]: agreed.  Please remove isdebugenable in the file

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java [hunk 3]
```diff
@@ -67,47 +67,52 @@ public Credential validate(Credential credential, RequestData data) throws WSSec
         if (credential == null || credential.getTimestamp() == null) {
             throw new WSSecurityException(ErrorCode.FAILURE, "noCredential");
         }
+
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java]: remove extra line

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java [hunk 4]
```diff
@@ -67,47 +67,52 @@ public Credential validate(Credential credential, RequestData data) throws WSSec
...
 
-        boolean timeStampStrict;
+        CONNECTTimestamp timeStamp = new CONNECTTimestamp(credential.getTimestamp());
+
+        validate(timeStamp, data);
+
+        return credential;
+    }
+
+    private void validate(CONNECTTimestamp timeStamp, RequestData data) throws WSSecurityException {
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java]: consider to put javadoc in file

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java [hunk 5]
```diff
@@ -117,10 +122,10 @@ protected long getTimeStampTTL() {
     protected long getTimeStampFutureTTL() {
         try {
             return PropertyAccessor.getInstance().getPropertyLong(NhincConstants.GATEWAY_PROPERTY_FILE,
-                    NhincConstants.TIMESTAMP_FUTURE_TIME_TO_LIVE);
+                NhincConstants.TIMESTAMP_FUTURE_TIME_TO_LIVE);
         } catch (PropertyAccessException pae) {
             LOG.warn("Failed to determine timestamp future TTL in gateway.properties, will use default value: {}",
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java]: consider to use warn or trace. It doesn't make sense to log twice with different message

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java [hunk 6]
```diff
@@ -130,10 +135,10 @@ protected long getTimeStampFutureTTL() {
     protected boolean getTimeStampStrict() {
         try {
             return PropertyAccessor.getInstance().getPropertyBoolean(NhincConstants.GATEWAY_PROPERTY_FILE,
-                    NhincConstants.TIMESTAMP_STRICT);
+                NhincConstants.TIMESTAMP_STRICT);
         } catch (PropertyAccessException pae) {
             LOG.warn("Failed to determine timestamp strict in gateway.properties, will use default value: {}",
-                    pae.getLocalizedMessage());
+                pae.getLocalizedMessage());
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/TimestampValidator.java]: Please see comment above

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

**Comment by mhpnguyen** [comment]
Endorse for merging without comment

**Review by mhpnguyen — APPROVED** [review]

**Comment by sjarral112** [comment]
Tested for Code format, Code practice and simplicity, no further change required.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java [hunk 7]
```diff
@@ -89,38 +86,55 @@ public boolean isExpired(Date invocationDate) {
...
-        if (invocationDate == null) {
-            invocationDate = new Date();
-        }
+        Date date = invocationDate == null ? new Date() : invocationDate;
 
-        long invocationTime = invocationDate.getTime();
+        long invocationTime = date.getTime();
         if (futureTimeToLive > 0) {
-            invocationDate.setTime(invocationTime + futureTimeToLive * 1000);
+            date.setTime(invocationTime + futureTimeToLive * 1000);
```
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java]: I don't follow the logic here.  Wouldn't the setTime on line 93 always be overridden by setTime on line 103?

Comment (jsmith2) [reply Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java]: Nevermind.  The confusion is that the same variable is used to represent two values, the value of the future time (so it's not created in the future) and of the past time (so it's confirmed that it isn't created too far in the past.  We might want to clean this up for clarity and appropriate variable naming.  Using a variable name like "date" isn't very clear.

**Review by jsmith2 — APPROVED** [review]

**Pull request merged**